packages/zod-openapi/src/index.ts

```diff
@@ -323,9 +323,18 @@ export class OpenAPIHono<
   }
 }
 
+type RoutingPath<P extends string> = P extends `${infer Head}/{${infer Param}}${infer Tail}` ? `${Head}/:${Param}${RoutingPath<Tail>}` : P
+
 export const createRoute = <P extends string, R extends Omit<RouteConfig, 'path'> & { path: P }>(
   routeConfig: R
-) => routeConfig
+) => {
+  return {
+    ...routeConfig,
+    getRoutingPath(): RoutingPath<R['path']> {
```

Comment:
The path is obtained from R, which can be obtained dynamically, because P is not determined dynamically from route.

Specifically, if P is not explicitly declared, it becomes a string.
[When type is calculated from P](https://www.typescriptlang.org/play?target=8#code/JYWwDg9gTgLgBDAnmApnA3nAShArjFAYQgDsAzYAczgF84yoIQ4ByAAQEMBnA7iAG3zBSXAPQAvCABMAtDAgyIqEhzDAWAKA1JU2PDGAlKABQ4wAFgB5jcFAA8CJKVzg8ohygD44AXjg37R2c4AAMAEnRDMhQoOAAJFA4pGlF0CKiY-w4oDhAaGnTyTIAVDmB+GhC4AH5QiISklIAuCNMcvIicISNTC0tS8s9KuCb-LXtIWDgAYxF4aahEgi6CXzhrWwcUJxc3DwAabE2glwB5EGAYSxWiUgpKQ5YwM3MWbwAyDDhni1GbGk8AAoNHA4Ix8LdyFRRlgNABKXzedAgsEoGC4KAkDAo0EAOnx4IIxChDxxcEoaJWHl65kBcJh+mpL2sSLJoMW6MxYP0kPuuJ+5lxizA-A40xQAEF+PxAaIADqpQG4gDU1ThKQerFELQAjCwEdw9N0TMzjJ4yTQUZbLRopChpqLFjM5q5puYUCAOKMOCREFpZiQeNyIWsFksUDdAcjQaCQGjzNJRiwKTAWPsyQLRgAiUS4LgxMRZ9Mx1EAR1wKB4o2jMee7S4oy4bo9HGLMZobfZlcggcr1bJACYAAxD-slksBxwwMfj0EsVQi4DTMzCEiiABWXFILBns5LTfdns7447bPbx5Ldqb7jABlISawaPcKAAbmgLGg8zE0xbj6eaHC-ougKayEiguIplSPQvHSGiiKIJYAHrVEAA)

[When type calculation is performed from R["path"]](https://www.typescriptlang.org/play?target=8#code/JYWwDg9gTgLgBDAnmApnA3nAShArjFAYQgDsAzYAczgF84yoIQ4ByAAQEMBnA7iAG3zBSXAPQAvCABMAtDAgyIqEhzDAWAKA1JU2PDGAlKABQ4wAFgB5jcFAA8CJKVzg8ohygD44AXjg37R2c4AAMAEnRDMhQoOAAJFA4pGlF0CKiY-w4oDhAaGnTyTIAVDmB+GhC4AH5QiISklIAuCNMcvIicISNTC0tS8s9KuCb-LXtIWDgAYxF4aahEgi6CXzhrWwcUJxc3DwAabE2glwB5EGAYSxWiUgpKQ5YwM3MWbwAyDDhni1GbGk8AAoNHA4Ix8LdyFRRlgNABKXzedAgsEoGC4KAkDAo0EAOnx4IIxChDxxcEoaJWHl65kBcJh+mpL2uAG0AEQ-cxsgC6SLJoMW6MxYP0kPuuM5uMWYH4HGmKAAgvx+IDRAAdVKA3EAamqcJSD1YohaAEYWAjuHpuiZmcZPGSaCjHY6NFIUNNZYsZnNXNNzCgQBxRhwSIgtLMSDwRRC1gsligboDkaDQSA0eZpKMWBSYCx9mTOaM2aJcFwYmI2fmU6iAI64FA8UbJlPPdpcUZcP0BjhVlM0XsChuQSMNptkgBMAAZJ2Pq9WI44YLO56CWKoZcBpmZhCRRAArLikFjLlfVzv+wMDuf9-l9q-Vt2d9xgAykLNYNHuFAANzQFjQpYxHmDpXjeNBwuGPqcmshIoLiOZUj0Lx0hooiiNWAB61RAA)

Reply:
Nice!